### PR TITLE
Fix publish to Sonatype in snapshot

### PR DESCRIPTION
**Pull request by nulls**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -101,6 +113,31 @@ jobs:
           flags: unittests
           fail_ci_if_error: false # optional (default = false)
 
+      - name: 'Publish a snapshot to GitHub'
+        id: publish-github
+        if: ${{ github.event_name == 'push' && github.ref_type == 'branch' && github.ref == 'refs/heads/main' && matrix.os == 'macos-latest' }}
```
Comment (sanyavertolet) [inline .github/workflows/build_and_test.yml:118]: why `macos-latest`?

**Review by sanyavertolet — APPROVED** [review]
Do we build the lib for `macos-arm64`?

**Comment by nulls** [comment]
> Do we build the lib for `macos-arm64`?

will be later. also the rest targets too

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:118]: as I remember `macos-latest` builds all required native targets.
but I'm still experimenting on it

**Pull request merged**